lib/zinnia.js

```diff
@@ -221,11 +221,20 @@ export async function run ({
   }
 
   let shouldRestart
+  const controller = new AbortController()
+  const { signal } = controller
 
   await Promise.all([
```

Comment:
The list of promises we are running in parallel is getting rather long. I find it difficult to understand the big picture because there are too many low-level implementation details obscuring the vision.

It would be great to refactor this code, extract smaller functions and simplify the `Promise.all` statement into something like this:

```js
await Promise.all([
  runModuleUpdater(),
  runContractAddressUpdater(),
  runChildProcesses(),
  waitForChildrenToClose(),
])
```

Such refactoring is out of scope of this PR, though.

Reply:
fantastic idea. I'll follow up!

Reply:
https://github.com/filecoin-station/core/pull/358